src/riak_kv_app.erl

```diff
@@ -49,7 +49,21 @@
 start(_Type, _StartArgs) ->
     riak_core_util:start_app_deps(riak_kv),
 
-    FSM_Limit = app_helper:get_env(riak_kv, fsm_limit, ?DEFAULT_FSM_LIMIT),
+    FSM_Limit = case app_helper:get_env(riak_kv, fsm_limit, ?DEFAULT_FSM_LIMIT) of
+                    undefined ->
+                        %% If it's explicitly set to 'undefined', that means "no limit".
+                        %% sidejob doesn't have an option to specify "infinity" but if
+                        %% we make the limit equal to the node's global process limit,
+                        %% then we're pretty much guaranteed to never exceed that:
+                        erlang:system_info(process_limit);
```

Comment:
I can definitely see your point here, but I think I would object to that change on principle. If we're going to allow people to say "never trigger overload no matter what", it seems like we ought to honor that, even if 99.9% of the time it's a stupid thing to do.